src/responsibleai/rai_analyse/generate_pdf.py

```diff
@@ -0,0 +1,1226 @@
+                    _class="header_cell"
+                ),
+                td(
+                    cm["fp"],
+                    _class="cell",
+                    _style="font-size:22px; color:#A80000; text-align: center;"
+                ),
+                td(
+                    p(
+                        cm["tn"],
+                        _style="font-size:22px; color:#107C10; text-align: center;"
+                    ),
+                    p(
+                        "correct prediction",
+                        _style="font-size:14px; text-align: center;"
+                    ),
+                    _class="cell"
+                )
+
+            ),
+            _style="width: 5in"
+        )
+
+    @staticmethod
+    def get_model_performance_page(data):
+
+        left_metric_elems = [p("Observe evidence of your model performance here:")]
+
+        def get_metric_bar_plot(mname, data):
+            y_0_filtermap = [True if i == data["classes"][0] else False for i in data["y_test"]]
+            y_1_filtermap = [True if i == data["classes"][1] else False for i in data["y_test"]]
+            class_0_metric = get_metric(mname, data["y_pred"][y_0_filtermap], data["y_test"][y_0_filtermap])
+            class_1_metric = get_metric(mname, data["y_pred"][y_1_filtermap], data["y_test"][y_1_filtermap])
+
+            y_data = ["Acutal \"{}\"".format(data["classes"][0]),
+                      "Acutal \"{}\"".format(data["classes"][1])]
+            x_data = [int(class_0_metric*100), int(class_1_metric*100)]
+            x_data = [[x, 100-x] for x in x_data]
+
+            legend = [m]
+            tickvals = [0, 25, 50, 75, 100]
+            ticktext = [str(x) + "%" for x in tickvals]
+
+            png_base64 = get_bar_plot(y_data, x_data, legend=legend,
+                                      tickvals=tickvals, ticktext=ticktext, tickappend="%")
+            return div(img(_src="data:image/png;base64,{}".format(png_base64)), _class="image_div")
+
+        main_elems = []
+        main_elems.append(ClassificationComponents._get_confusion_matrix_grid(data))
+
+        for m in data["metrics"]:
+            left_metric_elems.append(
+                ClassificationComponents._get_model_performance_explanation_text(m, data)
+            )
+            main_elems.append(
+                get_metric_bar_plot(m, data)
+            )
+
+        left_container = div(left_metric_elems, _class="left")
+        main_container = div(main_elems, _class="main")
+        return str(div(get_page_divider("Model Performance"), left_container, main_container, _class="container"))
+
+    @staticmethod
+    def get_cohorts_page(data, metrics_config):
+        return get_cohorts_page(data, metrics_config)
+
+    @staticmethod
+    def get_feature_importance_page(data):
+        return get_feature_importance_page(data)
+
+    @staticmethod
+    def get_causal_page(data):
+        return get_causal_page(data)
```

Comment:
Am I missing something? This looks recursive with no termination condition.

Reply:
the staticmethod is referenced by RegressionComponents.get_causal_page, get_causal_page references the common method defined at the top of the file

Reply:
refactor the code and split the file into common, classification and regression components. This ambiguity should be removed now.